src/mon/OSDMonitor.cc

```diff
@@ -280,8 +282,130 @@ OSDMonitor::OSDMonitor(
    inc_osd_cache(g_conf()->mon_osd_cache_size),
    full_osd_cache(g_conf()->mon_osd_cache_size),
    has_osdmap_manifest(false),
+   inc_cache(new IncCache(this)),
+   full_cache(new FullCache(this)),
    mapper(mn->cct, &mn->cpu_tp)
-{}
+{
+  cct->_conf.add_observer(this);
+  int r = _set_cache_sizes();
+  if (r < 0) {
+    derr << __func__ << " using default osd cache size - mon_osd_cache_size ("
+         << g_conf()->mon_osd_cache_size
+         << ") without priority cache management"
+         << dendl;
+  }
+}
+
+const char **OSDMonitor::get_tracked_conf_keys() const
+{
+  static const char* KEYS[] = {
+    "mon_memory_target",
+    "mon_osd_cache_size_min",
+    "mon_memory_autotune",
+    "rocksdb_autotune_cache_size",
+    NULL
+  };
+  return KEYS;
+}
```

Comment:
Good job on implementing an observer for the changes!  We need to do that on the osd/bluestore side as well.  It would be great if you could test/verify that this is working and implement the same thing there! :+1:

Reply:
Apparently, even though I set the flag for these options to FLAG_RUNTIME, I get the following message when I execute the command to change the config option,

``
$ sudo bin/ceph tell osd.* injectargs '--mon_memory_target=1073741824'
*** DEVELOPER MODE: setting PATH, PYTHONPATH and LD_LIBRARY_PATH ***
2019-06-06 19:41:07.282 7f4e54780700 -1 WARNING: all dangerous and experimental features are enabled.
2019-06-06 19:41:07.340 7f4e54780700 -1 WARNING: all dangerous and experimental features are enabled.
osd.0: mon_memory_target = '1073741824' (not observed, change may require restart) 
osd.1: mon_memory_target = '1073741824' (not observed, change may require restart) 
osd.2: mon_memory_target = '1073741824' (not observed, change may require restart)
``

Am I missing something here?

Reply:
```ceph tell injectargs osd.*``` is setting the option on the osds, where it has no effect. Since mimic you can also use ```ceph config set``` instead of ```ceph tell``` since the configuration is centralized in the monitors now.

Reply:
I was able to successfully test this using `ceph config set`. Thanks!